sdk/src/contexts/solana/context.ts

```diff
@@ -928,10 +935,28 @@ export class SolanaContext<
     );
     const recipientChainId = this.context.toChainId(recipientChain);
     const nonce = createNonce().readUint32LE();
+    const transaction = new Transaction();
     let transferIx: TransactionInstruction;
     if (token === NATIVE || token.chain === SOLANA_CHAIN_NAME) {
       const mint = token === NATIVE ? NATIVE_MINT : token.address;
       const wrapToken = token === NATIVE;
+      if (wrapToken) {
+        const ata = getAssociatedTokenAddressSync(
+          NATIVE_MINT,
+          new PublicKey(senderAddress),
+        );
+        try {
+          await getAccount(this.connection, ata);
+        } catch (e) {
+          const createAccountInst = createAssociatedTokenAccountInstruction(
```

Comment:
Would it be possible of adding a check to make sure the error is due to an inexistent ATA and not some other thing?

Reply:
I agree that this would be more robust. Unfortunately I don't think there's an error code to check against, but we can compare against the error string

Reply:
We could check `e instanceof TokenAccountNotFoundError` but I think that could have issues if the error were thrown from different libraries, going to leave it for now